smart-contracts/contracts/PublicLock.sol

```diff
@@ -221,20 +221,42 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
   }
 
   /**
-   * @dev Called by owner to wiwthdraw all funds from the lock.
+   * @dev Called by owner to withdraw all funds from the lock.
    * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
-   * TODO: consider partial withdraws?
    * TODO: check for re-entrency?
    */
-  function withdraw(
-  )
+  function withdraw()
     external
     onlyOwner
   {
     uint256 balance = address(this).balance;
     require(balance > 0, "Not enough funds");
+    _withdraw(balance);
```

Comment:
I understand we are not too focused on gas costs ATM, however a possible minor optimization here.  We are currently checking the balance twice and technically it could be done only once.

Reply:
Yeah, there was another spot in `_purchaseFor` where we were reading `keyprice` from storage 3 or 4 times...I modified to read it into memory once and then use that instead.
Can you elaborate on how we can eliminate 1 check here? I think it's worth it, and may as well do it now while I'm in there tinkering:)

Reply:
Oh sorry, I misread.  I thought _withdraw had a require line as well.

But on the topic of gas, a common practice in our .sol files today is to use temp variables.  I agree this is good for readability, but the Solidity compiler does not optimize that away.

This implementation is cheaper:
```
  function _withdraw(uint _amount)
    public
  {
    msg.sender.transfer(_amount);
  }
```

than the current implementation
```
  function _withdraw(uint _amount)
    public
  {
    address owner = msg.sender;
    owner.transfer(_amount);
  }
```

Removing the temp saves 13 gas.  Very little, but there are a few instances where we could do the same.

Reply:
Right, I see what you're saying. In our`_withdraw` function, as we're not transferring to `msg.sender` but to `owner`,  I think we could do this:
```
function _withdraw(uint _amount)
    private
  {
    Ownable.owner().transfer(_amount);
    emit Withdrawal(msg.sender, _amount);
  }
```

Reply:
cool. Done.